### PR TITLE
Fail prowjobs on UnprocessableEntity errors

### PR DESCRIPTION
**Pull request by kargakis**
Fixes https://github.com/kubernetes/test-infra/issues/4550

/assign spxtr

### CODE REVIEW

File: prow/plank/controller.go [hunk 1]
```diff
@@ -236,10 +236,17 @@ func (c *Controller) syncPendingJob(pj kube.ProwJob, pm map[string]kube.Pod, rep
...
 		id, pn, err := c.startPod(pj)
 		if err != nil {
-			return fmt.Errorf("error starting pod: %v", err)
+			_, isUnprocessable := err.(kube.UnprocessableEntityError)
+			if !isUnprocessable {
+				return fmt.Errorf("error starting pod: %v", err)
+			}
+			pj.Status.State = kube.ErrorState
+			pj.Status.CompletionTime = time.Now()
+			pj.Status.Description = "Job cannot be processed."
```
Comment (spxtr) [inline prow/plank/controller.go]: We should include the error reason as well. The API server gives a handy message like "job name blabla-123.5 is not a valid DNS name" or something that we should see.

Comment (kargakis) [reply prow/plank/controller.go]: Thought about it, just not sure how will it show up on Github. I feel though that the reports package is not meant to handle reports only to Github.

Comment (spxtr) [reply prow/plank/controller.go]: We can log it as a warning to start out with. Maybe it's worth putting a "reason" string in the ProwJob struct that can be exposed by deck, but for now I'm happy with just logging it.

Comment (kargakis) [reply prow/plank/controller.go]: Logging SGTM. Updated.

File: prow/plank/controller.go [hunk 2]
```diff
@@ -237,10 +237,18 @@ func (c *Controller) syncPendingJob(pj kube.ProwJob, pm map[string]kube.Pod, rep
...
+			logrus.WithField("job", pj.Spec.Job).Warningf("%v", err)
```
Comment (spxtr) [inline prow/plank/controller.go]: `logrus.WithField("job", pj.Spec.Job).WithError(err).Warning("Unprocessable pod.")`

File: prow/plank/controller.go [hunk 3]
```diff
@@ -317,22 +325,33 @@ func (c *Controller) syncNonPendingJob(pj kube.ProwJob, pm map[string]kube.Pod,
...
 		id, pn, err = c.startPod(pj)
 		if err != nil {
-			return fmt.Errorf("error starting pod: %v", err)
+			_, isUnprocessable := err.(kube.UnprocessableEntityError)
+			if !isUnprocessable {
+				return fmt.Errorf("error starting pod: %v", err)
+			}
+			pj.Status.State = kube.ErrorState
+			pj.Status.CompletionTime = time.Now()
+			pj.Status.Description = "Job cannot be processed."
```
Comment (spxtr) [inline prow/plank/controller.go]: Also log here.

**Comment by spxtr** [comment]
/lgtm

Minor nits.

File: prow/plank/controller.go (hunk 2 above)
Comment (kargakis) [reply prow/plank/controller.go]: ok

File: prow/plank/controller.go (hunk 3 above)
Comment (kargakis) [reply prow/plank/controller.go]: ok

**Pull request merged**